### PR TITLE
Implemented CMBk

### PR DESCRIPTION
**Pull request by carlosggarcia**
I have implemented the CMBk tracer for S4 using the noise from https://github.com/toshiyan/cmblensplus/blob/master/example/data/kappa_noise_cmbs4.tar as indicated in https://cmb-s4.uchicago.edu/wiki/index.php/Survey_Performance_Expectations#Lensing_reconstruction_noise_curve. 

The noise used is the inverse of the sum of the inverse of the different contributions; i.e. ` nl = 1 / np.sum(1 / file_loaded[1: 8], axis=0)`

### CODE REVIEW

File: data/datagen.py [hunk 1]
```diff
@@ -130,15 +150,16 @@ def _get_nls(self):
...
+        nls = np.zeros([self.n_tot, self.n_tot, ll['n_bpw']])
         sgamma = self.c.get('e_rms', 0.28)
         # Clustering first
         for i in range(self.n_cl):
             nls[i, i, :] = 1./self.ndens_cl[i]
         # Then shear
         for i in range(self.n_sh):
             nls[i+self.n_cl, i+self.n_cl, :] = sgamma**2/self.ndens_sh[i]
+        # Then CMBk
+            nls[-1, -1, :] = self._get_cmbk_nl()[1]
```
Comment (damonge) [inline data/datagen.py:162]: extra tab?

**Review by damonge — CHANGES_REQUESTED** [review]
OK, great! Just one question: have we checked that the noise curves are for CMB lensing convergence? Could you make a quick plot showing the CMB lensing autospectrum and the corresponding N_l curve from S4 just to make sure?

File: data/datagen.py (hunk 1 above)
Comment (carlosggarcia) [reply data/datagen.py]: Fixed

**Comment by carlosggarcia** [comment]
@damonge  thanks for having a look. Here it's the plot. Also, I fixed a bug with the data types for the tracer combinations that involved cmbk image

File: data/datagen.py [hunk 2]
```diff
@@ -128,11 +128,14 @@ def _get_cl_type(self, i, j):
...
                 return 'cl_0e'
-        else:
-            if (j < self.n_cl) or (i == self.n_tot - 1):
+        elif i < self.n_cl + self.n_sh:
+            if (j < self.n_cl) or (j == self.n_tot - 1):
                 return 'cl_0e'
             else:
                 return 'cl_ee'
+        else:
+            # Else = cmbk
```
Comment (damonge) [inline data/datagen.py:137]: don't you need to check j here?

**Review by damonge — CHANGES_REQUESTED** [review]
OK, I think the plot looks right. Just one final comment.

File: data/datagen.py (hunk 2 above)
Comment (carlosggarcia) [reply data/datagen.py:137]: I've added it just in case but it returned the correct data type since it was called always so that j >= i.

**Review by damonge — APPROVED** [review]

File: data/datagen.py (hunk 2 above)
Comment (damonge) [reply data/datagen.py:137]: yeah, just in case we change this in the future. Thanks!

Comment (carlosggarcia) [reply data/datagen.py:137]: if this becomes more complicated I would rather update sacc and use `s.get_data_types(tracers=(tr1, tr2))`

**Pull request merged**